src/BenchmarkDotNet.Core/Diagnosers/MemoryDiagnoser.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using BenchmarkDotNet.Columns;
+using BenchmarkDotNet.Environments;
+using BenchmarkDotNet.Loggers;
+using BenchmarkDotNet.Reports;
+using BenchmarkDotNet.Running;
+using System.Linq;
+using BenchmarkDotNet.Engines;
+
+namespace BenchmarkDotNet.Diagnosers
+{
+    public class MemoryDiagnoser : IDiagnoser
+    {
+        public static readonly MemoryDiagnoser Default = new MemoryDiagnoser();
+
+        private readonly Dictionary<Benchmark, GcStats> results = new Dictionary<Benchmark, GcStats>();
+
+        public IColumnProvider GetColumnProvider() => new SimpleColumnProvider(
+            new GCCollectionColumn(results, 0),
+            new GCCollectionColumn(results, 1),
+            new GCCollectionColumn(results, 2),
+            new AllocationColumn(results));
+
+        // the methods are left empty on purpose
+        // the action takes places in other process, and the values are gathered by Engine
+        public void BeforeAnythingElse(Process process, Benchmark benchmark) { }
+        public void AfterSetup(Process process, Benchmark benchmark) { }
+        public void BeforeCleanup() { }
+
+        public void ProcessResults(Benchmark benchmark, BenchmarkReport report)
+        {
+            results.Add(benchmark, report.GcStats);
+        }
+
+        public void DisplayResults(ILogger logger) { }
+
+        public class AllocationColumn : IColumn
+        {
+            private readonly Dictionary<Benchmark, GcStats> results;
+
+            public AllocationColumn(Dictionary<Benchmark, GcStats> results)
+            {
+                this.results = results;
+            }
+
+            public string Id => nameof(AllocationColumn);
+            public string ColumnName => "Bytes Allocated/Op";
+            public bool IsDefault(Summary summary, Benchmark benchmark) => false;
+            public bool IsAvailable(Summary summary) => true;
+            public bool AlwaysShow => true;
+            public ColumnCategory Category => ColumnCategory.Diagnoser;
+            public int PriorityInCategory => 0;
+
+            public string GetValue(Summary summary, Benchmark benchmark)
+            {
+#if !CORE
+                if (results.ContainsKey(benchmark))
+                {
+                    var result = results[benchmark];
+                    // TODO scale this based on the minimum value in the column, i.e. use B/KB/MB as appropriate
```

Comment:
You can probably remove this comment now, I don't think there's any point in scaling it. 

Bytes are the logical choice for the per/ops measurement and it might get confusing to sometimes show KB or MB instead.